src/test/java/org/apache/datasketches/CrossCheckQuantilesTest.java

```diff
@@ -0,0 +1,356 @@
+    //One for each value in the stream + one inbetween and zero
+    int numTR = 2 * N + 1;
+    double[] testRanks = new double[numTR];
+    testRanks[0] = 0;
+    for (int i = 1; i < numTR; i++) {
+      testRanks[i] = (double) i / (numTR - 1);
+    }
+
+    //TEST RESULT ARRAYS, NI = non inclusive, I = inclusive
+    if (testRankResults_NI == null) {
+      testRankResults_NI = new double[numTV];
+      testRankResults_I  = new double[numTV];
+      testQuantileFResults_NI = new float[numTR];
+      testQuantileFResults_I  = new float[numTR];
+      testQuantileDResults_NI = new double[numTR];
+      testQuantileDResults_I  = new double[numTR];
+    }
+
+    println("Sketch Input with exact weights and ranks:");
+    println("  Sketch only keeps individual value weights per level.");
+    println("  Cumulative Weights are computed in Sorted View.");
+    println("  Normalized Ranks are computed on the fly.");
+    println("");
+    printf("%16s%16s%16s\n", "Value", "CumWeight", "NormalizedRank");
+
+    //LOAD THE SKETCHES and PRINT
+    for (int i = 0; i < N; i++) {
+      printf("%16.1f%16d%16.3f\n", skFValues[i], i + 1, (i + 1.0)/N);
+      reqSk.update(skFValues[i]);
+      kllSk.update(skFValues[i]);
+      udSk.update((skDValues[i]));
+    }
+    println("");
+
+    //REQ SORTED VIEW DATA:
+    ReqSketchSortedView rssv = null;
+    if (skType.toString().startsWith("REQ")) {
+      rssv = new ReqSketchSortedView(reqSk);
+      println(rssv.toString(1, 16));
+    }
+
+    /**************************************/
+
+    println(skType.toString() + " GetQuantile(NormalizedRank), INCLUSIVE = " + inclusive);
+    println("  CumWeight is for illustration");
+    println("  Convert NormalizedRank to CumWeight (CW).");
+    println("  Search RSSV CumWeights[] array:");
+    println("    Non Inclusive (uses GT): arr[A] <= CW <  arr[B], return B");
+    println("    Inclusive     (uses GE): arr[A] <  CW <= arr[B], return B");
+    println("  Return Values[B]");
+    println("");
+    printf("%16s%16s%16s%16s\n", "NormalizedRank", "CumWeight", "Quantile", "True_Quantile");
+    for (int i = 0; i < numTR; i++) {
+      double testRank = testRanks[i];
+      float qF; //float result
+      double qD;//double result
+      switch (skType) {
+        case REQ: { //this case creates the expected values for all the others
+          qF = reqSk.getQuantile(testRank, inclusive);
+          qD = qF;
+          if (inclusive) {
+            testQuantileFResults_I[i] = qF;
+            testQuantileDResults_I[i] = qD;
+          }
+          else {
+            testQuantileFResults_NI[i] = qF;
+            testQuantileDResults_NI[i] = qD;
+          }
+          break;
+        }
+        case REQ_SV: {
+          qF = rssv.getQuantile(testRank, inclusive);
+          qD = 0;
+          if (inclusive) { assertEquals(qF, testQuantileFResults_I[i]); }
+          else { assertEquals(qF, testQuantileFResults_NI[i]); };
+          break;
```

Comment:
I only have one additional question for this review regarding the intent of these tests - is it fair to say that the new RSSV is compared to the existing quantiles sketches and the output checked for equality?
If so, what is the purpose behind the standalone sorted view if the quantiles can be retrieved directly from the sketch?

Reply:
The Sorted View provides a view of the data retained by the sketch that would be cumbersome to get any other way.  One can iterate of the contents of the sketch, but the result is not sorted.  Trying to use getQuantiles would be very cumbersome since one doesn't know what ranks to use to supply the getQuantiles method.  Even worse, suppose it is a large sketch that has retained 1000 values from a stream of millions (or billions).   One would have to execute the getQuantiles method many thousands of times, and using trial & error, try to figure out what the sketch actually has retained.    

The data from a Sorted view is an unbiased sample of the input stream that can be used for other kinds of analysis not directly provided by the sketch.  A good example comparing two sketches using the Kolmogorov-Smirnov test.  One needs this sorted view for the test.

This sorted view can also be used for multiple getRank and getQuantile queries once it has been created. Because it takes some computational work to create this sorted view, it doesn't make sense to create this sorted view just for single getRank queries.  For the first getQuantile queries, it must be created. But for all queries after the first, assuming the sketch has not been updated, the getQuantile and getRank queries are very fast.

Reply:
This comment has been added to the class javadoc for all the quantile sketches' Sorted Views.